### PR TITLE
Adds `Publisher.setOutputType(to:)`.

### PR DESCRIPTION
**Pull request by jasdev**
h/t @sharplet for teaching me about this one. :nodg:

### CODE REVIEW

File: Sources/Operators/SetOutputType.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//  Created by Jasdev Singh on 02/04/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher where Output == Never {
+    /// An output analog to [Publisher.setFailureType(to:)](https://developer.apple.com/documentation/combine/publisher/3204753-setfailuretype) for when `Output == Never`. This is especially helpful when chained after [.ignoreOutput()](https://developer.apple.com/documentation/combine/publisher/3204714-ignoreoutput) operator calls.
+    ///
+    /// - Parameter outputType: The new output type for downstream.
```
Comment (freak4pc) [inline Sources/Operators/SetOutputType.swift:14]:
```suggestion
    /// - parameter outputType: The new output type for downstream.
```

File: Sources/Operators/SetOutputType.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    ///
+    /// - Returns: A publisher with a `NewOutput` output type.
```
Comment (freak4pc) [inline Sources/Operators/SetOutputType.swift:16]:
```suggestion
    /// - returns: A publisher with a `NewOutput` output type.
```

**Comment by freak4pc** [comment]
I see you added the fixes but still haven't answered about the use-case question ... looking forward to your answer 🙏

**Comment by jasdev** [comment]
> I see you added the fixes but still haven't answered about the use-case question ... looking forward to your answer 🙏

Yep, sorry—just haven’t had the time. I’ll try to get to this when I can! 🙇🏽‍♂️

**Comment by jasdev** [comment]
Thank you for asking the question that sparked the post! It’s a surprisingly nuanced operator 😄 :nodg:

**Pull request merged**